src/contract-abi.ts

```diff
@@ -0,0 +1,336 @@
+    } else if (val === 'int128') {
+      return { id: ClarityAbiTypeId.ClarityAbiTypeInt128, type: val };
+    } else if (val === 'bool') {
+      return { id: ClarityAbiTypeId.ClarityAbiTypeBool, type: val };
+    } else if (val === 'principal') {
+      return { id: ClarityAbiTypeId.ClarityAbiTypePrincipal, type: val };
+    } else if (val === 'none') {
+      return { id: ClarityAbiTypeId.ClarityAbiTypeNone, type: val };
+    } else {
+      throw new Error(`Unexpected Clarity ABI type primitive: ${JSON.stringify(val)}`);
+    }
+  } else if (isClarityAbiBuffer(val)) {
+    return { id: ClarityAbiTypeId.ClarityAbiTypeBuffer, type: val };
+  } else if (isClarityAbiResponse(val)) {
+    return { id: ClarityAbiTypeId.ClarityAbiTypeResponse, type: val };
+  } else if (isClarityAbiOptional(val)) {
+    return { id: ClarityAbiTypeId.ClarityAbiTypeOptional, type: val };
+  } else if (isClarityAbiTuple(val)) {
+    return { id: ClarityAbiTypeId.ClarityAbiTypeTuple, type: val };
+  } else if (isClarityAbiList(val)) {
+    return { id: ClarityAbiTypeId.ClarityAbiTypeList, type: val };
+  } else {
+    throw new Error(`Unexpected Clarity ABI type: ${JSON.stringify(val)}`);
+  }
+}
+
+function encodeClarityValue(type: ClarityAbiType, val: string): ClarityValue;
+function encodeClarityValue(type: ClarityAbiTypeUnion, val: string): ClarityValue;
+function encodeClarityValue(
+  input: ClarityAbiTypeUnion | ClarityAbiType,
+  val: string
+): ClarityValue {
+  let union: ClarityAbiTypeUnion;
+  if ((input as ClarityAbiTypeUnion).id !== undefined) {
+    union = input as ClarityAbiTypeUnion;
+  } else {
+    union = getTypeUnion(input as ClarityAbiType);
+  }
+  switch (union.id) {
+    case ClarityAbiTypeId.ClarityAbiTypeUInt128:
+      return uintCV(val);
+    case ClarityAbiTypeId.ClarityAbiTypeInt128:
+      return intCV(val);
+    case ClarityAbiTypeId.ClarityAbiTypeBool:
+      if (val === 'false' || val === '0') return falseCV();
+      else if (val === 'true' || val === '1') return trueCV();
+      else throw new Error(`Unexpected Clarity bool value: ${JSON.stringify(val)}`);
+    case ClarityAbiTypeId.ClarityAbiTypePrincipal:
+      if (val.includes('.')) {
+        const [addr, name] = val.split('.');
+        return contractPrincipalCV(addr, name);
+      } else {
+        return standardPrincipalCV(val);
+      }
+    case ClarityAbiTypeId.ClarityAbiTypeNone:
+      return noneCV();
+    case ClarityAbiTypeId.ClarityAbiTypeBuffer:
+      return bufferCV(Buffer.from(val, 'utf8'));
+    case ClarityAbiTypeId.ClarityAbiTypeResponse:
+      throw new NotImplementedError(`Unsupported encoding for Clarity type: ${union.id}`);
+    case ClarityAbiTypeId.ClarityAbiTypeOptional:
+      throw new NotImplementedError(`Unsupported encoding for Clarity type: ${union.id}`);
+    case ClarityAbiTypeId.ClarityAbiTypeTuple:
+      throw new NotImplementedError(`Unsupported encoding for Clarity type: ${union.id}`);
```

Comment:
Should we try to support parsing tuple and list inputs? Seems useful but the syntax isn't clear to me and seems like a shell escaping nightmare

Reply:
Remind me how this function is being used in the sidecar?

I'd be down to try and implement this (probably in a later PR), but I agree, I think it might be tricky...

Reply:
Sidecar doesn't support parsing any complicated types like this. +1 for doing in a later PR